tidy3d/plugins/design/method.py

```diff
@@ -23,335 +20,780 @@ class Method(Tidy3dBaseModel, ABC):
+        for param, convert in param_converter.items():
+            for sol in sol_dict_list:
+                if isinstance(sol[param], float):
+                    sol[param] = int(round(sol[param], 0))
+                sol[param] = convert[sol[param]]
 
-    Example
-    -------
-    >>> import tidy3d.plugins.design as tdd
-    >>> method = tdd.MethodMonteCarlo(num_points=20)
+
+class MethodBayOpt(MethodOptimize, ABC):
+    """A standard method for performing a bayesian optimization search.
+
+    The fitness function is maximising by default.
     """
 
-    def get_sampler(self, parameters: Tuple[ParameterType, ...]) -> qmc.QMCEngine:
-        """Sampler for this ``Method`` class."""
+    initial_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of initial random search iterations",
+        description="The number of search runs to be done initialially with parameter values picked randomly. This provides a starting point for the Gaussian processor to optimize from. These solutions can be computed as a single ``Batch`` if the pre function generates ``Simulation`` objects.",
+    )
 
-        d = len(parameters)
-        return DEFAULT_MONTE_CARLO_SAMPLER_TYPE(d=d)
+    n_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of bayesian optimization iterations",
+        description="Following the initial search, this is number of iterations the Gaussian processor should be sequentially called to suggest parameter values and register the results.",
+    )
 
+    acq_func: Literal["ucb", "ei", "poi"] = pd.Field(
+        default="ucb",
+        title="Type of acquisition function",
+        description="The type of acquisition function that should be used to suggest parameter values. More detail available `here <https://bayesian-optimization.github.io/BayesianOptimization/exploitation_vs_exploration.html>`_",
+    )
 
-class MethodRandom(AbstractMethodRandom):
-    """Select sampling points uniformly at random.
+    kappa: pd.PositiveFloat = pd.Field(
+        default=2.5,
+        title="Kappa",
+        description="The kappa coefficient used by the ``ucb`` acquisition function. More detail available `here <https://bayesian-optimization.github.io/BayesianOptimization/exploitation_vs_exploration.html>`_",
+    )
 
-    Example
-    -------
-    >>> import tidy3d.plugins.design as tdd
-    >>> method = tdd.MethodRandom(num_points=20, monte_carlo_warning=False)
+    xi: pd.NonNegativeFloat = pd.Field(
+        default=0.0,
+        title="Xi",
+        description="The Xi coefficient used by the ``ei`` and ``poi`` acquisition functions. More detail available `here <https://bayesian-optimization.github.io/BayesianOptimization/exploitation_vs_exploration.html>`_",
+    )
+
+    def get_run_count(self, parameters: list = None) -> int:
+        return self.initial_iter + self.n_iter
+
+    def _run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable, console) -> Tuple[Any]:
```

Comment:
this method is a bit dense, but maybe that's ok.. I'm having trouble following so maybe either a mild refactor or some inline comments could help

Reply:
Yeah there's become a lot of fiddly code to get Bay opt succesfully through all the different scenarios. I've added some more comments and spacing.